xds/src/main/java/io/grpc/xds/RingHashLoadBalancer.java

```diff
@@ -207,6 +217,28 @@ private boolean validateAddrList(List<EquivalentAddressGroup> addrList) {
     return true;
   }
 
+  private String validateNoDuplicateAddresses(List<EquivalentAddressGroup> addrList) {
+    Set<SocketAddress> addresses = new HashSet<>();
+    Map<String, Integer> dups = new HashMap<>();
+    for (EquivalentAddressGroup eag : addrList) {
+      for (SocketAddress address : eag.getAddresses()) {
+        if (!addresses.add(address)) {
+          String addrStr = address.toString();
+          Integer oldCount = dups.computeIfAbsent(addrStr, (s) -> 0);
```

Comment:
`dups.getOrDefault(addrStr, 0)?`
Better yet - we could use guava's [Multiset](https://guava.dev/releases/18.0/api/docs/com/google/common/collect/Multiset.html) instead of the map - it's intended this exact case of counting objects.

Reply:
Using Multiset as suggested, thanks!